Date/Date.cpp

```diff
@@ -0,0 +1,63 @@
+#include <iostream>
+#include "Date.h"
+
+using namespace std;
+
+void Date::validate(int day, int month, int year) {
+    int daysInMonth[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+    if ( day > daysInMonth[month] || day < 1 ) {
+        throw InvalidDateException("Invalid Day!");
+    }
+    if ( month < 1 || month > 12 ) {
```

Comment:
You should do this validation first. In other case ArrayOutOfBoundException can be thrown if here: ` if ( day > daysInMonth[month] || day < 1 ) {`